WordPress/Classes/ViewRelated/Blog/BlogListViewController.m

```diff
@@ -218,6 +228,40 @@ - (void)maybeShowNUX
     }
 }
 
+- (void)maybeShowNoResultsView
+{
+    NSUInteger count = [self countForAllBlogs];
+    BOOL hasSites = (count != NSNotFound && count > 0);
+
+    // If we've gone from no results to having just one site, the user has
+    // added a new site so we should auto-select it
+    if (!self.noResultsView.hidden && count == 1) {
+        [self bypassBlogListViewController];
+    }
+
+    self.noResultsView.hidden = hasSites;
+
+    // If we have no results, set the split view to full width
+    WPSplitViewController *splitViewController = (WPSplitViewController *)self.splitViewController;
+    if ([splitViewController isKindOfClass:[WPSplitViewController class]]) {
+        splitViewController.dimsDetailViewControllerAutomatically = hasSites;
+        splitViewController.wpPrimaryColumnWidth = (hasSites) ? WPSplitViewControllerPrimaryColumnWidthNarrow : WPSplitViewControllerPrimaryColumnWidthFull;
+    }
+
+    // Nil out our blog details VC reference if the blog no longer exists
+    if (self.blogDetailsViewController && ![self.resultsController.fetchedObjects containsObject:self.blogDetailsViewController.blog]) {
+        self.blogDetailsViewController = nil;
+    }
```

Comment:
Something seems sketchy about having `blogDetailsViewController` and `splitViewController` logic within a method that is used for _maybe_ toggling the visibility of the view. Perhaps `bypassBlogListViewController` should be moved to wherever `maybeShowNoResultsView` itself is called. Or maybe this method should be renamed for a bit more context as to what is going on?